### DISCUSSION TITLE
Thinking about driver time lags in understanding the salt front dynamics

### DISCUSSION THREAD

**Discussion by galengorski** [USGS-R/drb-estuary-salinity-ml#48] [Ideas]
We're interested in the general question, "What are the key drivers of the salt front location and (how) do those vary as a function of the salt front location?"

One approach is to bin the salt front time series into "river mile bins" based on changes in river geometry or bathymetry and look at the relationship between drivers and salt front location within each one of those bins.

![salt_front_time_series_with_intervals](https://user-images.githubusercontent.com/32860978/153631159-fb081a10-7f14-47a7-ba2b-9b1e5fe8d446.png)

This is a time series of the salt front location with the red lines indicating the locations of the bins. The bottom panels show the discharge and specific conductivity at Trenton and Schuylkill. These are the drivers we'll start with.

**Comment by galengorski** [comment]
When we look at the correlation and mutual information between discharge and salt front location within each one of those river mile bins, it looks like this:

![discharge_associations](https://user-images.githubusercontent.com/32860978/153631611-5f36683f-8879-4f51-9f0b-f90609614970.png)

The top panel shows that in general there is a stronger correlation (more negative) when the salt front is lower in the estuary, and the correlation gets weaker as the salt front moves up. The bottom panel shows mutual information, which doesn't show much of a signal at any point in the estuary. Stronger correlation in the lower estuary make some sense as this is when discharge is likely more variable, in contrast when the salt front is further up the estuary, the discharge is low and doesn't change much.

**Comment by galengorski** [comment]
This led to the question of how does past discharge affect the movement of the salt front in each river mile bin. To answer that, I did the following:

1. For each day, take the mean and standard deviation of the previous n days of discharge
2. This results in a running mean (and standard deviation) of the previous n days of discharge
3. For each river mile bin, calculate the correlation and mutual information between the mean (and standard deviation) of the previous n days of discharge and the current salt front location
4. Do this for n = 1, 7, 15, 30, 60, 120 days to determine the "time scale" of influence for each driver at each river mile bin

Here is what it looks like for correlation with mean discharge at Trenton:

![trenton_discharge_windows_correlation](https://user-images.githubusercontent.com/32860978/153633243-1caa4df3-0eb7-4845-a145-d60ba5943354.png)

I think it's interesting to look at these plots column by column. For example, the last river mile bin 82-91 show that the mean discharge from the last 60 days is a much stronger predictor than the last 15 days. Each river mile bin shows distinct time scales of influence, lower river mile bins are correlated with mean discharge of the past 7-15 days.  

The plot below shows the same idea but for mutual information instead of correlation:

![trenton_discharge_windows_mutual_information](https://user-images.githubusercontent.com/32860978/153634022-5cd878b8-09bf-416e-8b6f-50076d7c6d33.png)

I think what is interesting here is that although the patterns are similar for correlation and mutual information, river mile bin 78-82 shows a very different relationship. This is an area of the river where it widens out and older water may become "trapped" during periods of low flow. This might indicate that there are non-linearities in the relationship between discharge and salt front here.

**Reply by salme146** [reply]
After meeting with Galen to discuss these results I looked at the COAWST modeled depth-averaged velocities at Trenton vs. observed discharge at Trenton (USGS) and plotted them:

![Modeled_Velocity_Trenton_Discharge_2019](https://user-images.githubusercontent.com/14081312/153670202-5ec539d8-ea75-4e33-a7c9-f5fe8b374631.jpg)

Given the river mile "reaches" I calculated the time scale in days for the water at Trenton to arrive at each reach and plotted the results here:

![Modeled_Timescale_Trenton_Discharge_2019](https://user-images.githubusercontent.com/14081312/153670797-b8d9f1b5-be51-4898-bd14-e9da7c068dee.jpg)

This makes sense, as the higher discharges correlate to higher flows, which means water from Trenton will arrive at the reach sooner, when flow is faster. In the summer months it takes longer for water to get to each of these locations, so the effect of discharge on the salinity in each location is lagged by a different amount.

**Reply by galengorski** [reply]
That's super interesting, these figures are awesome! What do you think about converting these to a travel time distribution for each river mile bin so that way for example R6: 82-91 has distribution of times for discharge to reach that area. We could then look at when the "time scales of influence" from the heatmaps above are significantly different from the mean travel time distribution and that could clue us into areas where discharge is more and less dominant in influencing the salt front...

**Reply by salme146** [reply]
If we take a look at the average time it takes water at Trenton to get to each "reach" : 

Annually
R1:31-58 : 6.5 days
R2:58-68 : 5.7 days
R3:68-70 : 5.5 days
R4:70-78 : 4.8 days
R5:78-82 : 4.5 days 
R6:82-91 : 3.7 days

August-October
R1:31-58 : 13.5 days (~Spring-Neap tidal cycle)
R2:58-68 : 11.7 days
R3:68-70 : 11.4 days
R4:70-78 : 10 days
R5:78-82 : 9.2 days 
R6:82-91 : 7.6 days

**Comment by galengorski** [comment]
Here are the same plots, but instead of looking at the mean of the previous n days of discharge at Trenton, I looked at the standard deviation of the previous n days of discharge at Trenton. 

Correlation:

![trenton_discharge_windows_correlation_sd](https://user-images.githubusercontent.com/32860978/153634795-0ffd0617-632c-40f4-9185-46fcfbb5f81c.png)

Mutual information:

![trenton_discharge_windows_mutual_information_sd](https://user-images.githubusercontent.com/32860978/153634840-ef35a01d-b807-40c2-8447-2249df6ca20b.png)

The patterns don't look all that different from looking at the mean, which confirms that periods of low and relatively uniform flow are associated with salt front movement in the upper estuary.

**Reply by galengorski** [reply]
![finescale_correlation_trenton_discharge](https://user-images.githubusercontent.com/32860978/154371319-d7aa99a8-a576-461d-a1d7-9305808c0e60.png)

This is a finer scaled version of the correlation heatmap, it looks like the heatmaps capture the overall patterns ok, but the finer scale shows more structure.

river mile: minimum correlations
31-58: 12
58-68: 12 
68-78: 33
70-78: 87
78-82: 21
82-91: 54

**Reply by galengorski** [reply]
![finescale_mutual_information_trenton_discharge](https://user-images.githubusercontent.com/32860978/154371913-a54b83a2-13cd-4456-b72d-6ea6eb6af301.png)

And here is a finer scaled version for the mutual information heatmap

river mile: maximum mutual information
31-58: 66
58-68: 16 
68-78: 10
70-78: 36
78-82: 82
82-91: 68

**Reply by salme146** [reply]
Noes from 2/17 meeting:

Different peaks in the data (say brown line) might correspond to different dynamics and time scales of those dynamics (large discharge events, and drought periods). So it might be useful to dig a little deeper into the timescales of influence. 

Sub-daily data - would have enough data to make calculations about binned flow. sub daily data would also lead towards using tidal data which is really important. daily tidal signals change on sub 6 hour time scales, spring-neap tidal time scales are 10-14 days.

**Comment by galengorski** [comment]
The plots for the Schuylkill look very similar, and in the interest of not clogging this discussion with plots, I'll leave those out.

For next steps, @salme146 suggested looking at the "expected time scale of influence" might be given the average river velocity and the distance from Trenton. Comparing the expected and actual time scales of influence might give us an idea of when discharge is dominant and when other drivers are more important.

**Reply by salme146** [reply]
@galengorski , I posted plots above looking at river velocity vs discharge for 2019 and then plotting travel times from Trenton to each of the reaches we picked for the analysis. I used the highest river mile to calculate the time scale. I am thinking if we are talking more about "influence" maybe I should use the middle of the reach. In any case this gives you an idea of time scales. They are shorter than I thought, but again, this is based on modeled velocities at Trenton.
